python/TestHarness/testers/CSVDiff.py

```diff
@@ -7,114 +7,258 @@
+        return output_dict
 
-            if self.specs.isValid('abs_zero'):
-                csvdiff.append('--abs-zero %s' % (self.specs['abs_zero']))
-
-            if self.specs.isValid('comparison_file'):
-                comparison_file = os.path.join(self.getTestDir(), self.specs['comparison_file'])
-                if os.path.exists(comparison_file):
-                    csvdiff.append('--comparison-file %s' % (comparison_file))
-                else:
-                    self.setStatus(self.fail, 'MISSING COMPARISON FILE')
-                    return commands
+    def processResults(self, moose_dir, options, output):
+        if self.specs.isValid('comparison_file'):
+            comparison_file = os.path.join(self.getTestDir(), self.specs['comparison_file'])
+            if not os.path.exists(comparison_file):
+                self.setStatus(self.fail, 'MISSING COMPARISON FILE')
+                return output
+            else:
+                self.custom_params = self.parseComparisonFile(comparison_file)
+        return SchemaDiff.processResults(self, moose_dir, options, output)
 
-            if self.specs.isValid('override_columns'):
-                csvdiff.append('--custom-columns %s' % (' '.join(self.specs['override_columns'])))
+    def do_deepdiff(self, orig, comp, rel_err, abs_err, abs_zero, exclude_values:list=None):
+        """ Perform DIFF Comparison and return (diff, error) tuple results """
+        (diff, error) = '', None
 
-            if self.specs.isValid('override_rel_err'):
-                csvdiff.append('--custom-rel-err %s' % (' '.join(self.specs['override_rel_err'])))
+        # Sanity check on column size. Test Spec options like custom_columns, ignore_columns
+        # have already been accounted for (see augment_csv method)
+        for column in orig:
+            if len(orig[column]) != len(comp[column]):
+                diff = f'Columns with header \'{column}\' are not the same length'
+                return (diff, error)
 
-            if self.specs.isValid('override_abs_zero'):
-                csvdiff.append('--custom-abs-zero %s' % (' '.join(self.specs['override_abs_zero'])))
+        # Override global params if comparison file is used
+        if self.custom_params:
+            rel_err = abs(float(self.custom_params.get('RELATIVE', self.specs['rel_err'])))
+            abs_err = abs(float(self.custom_params.get('ABSOLUTE', self.specs['abs_err'])))
+            abs_zero = abs(float(self.custom_params.get('ZERO', self.specs['abs_zero'])))
 
-            if self.specs.isValid('ignore_columns'):
-                csvdiff.append('--ignore-fields %s' % (' '.join(self.specs['ignore_columns'])))
+        # Set nested loop values so variables can be used for error printing
+        value_id = 0
+        value = ''
+        for index, column in enumerate(orig):
+            _rel_err = abs(rel_err)
+            _abs_err = abs(abs_err)
+            _abs_zero = abs(abs_zero)
```

Comment:
Why underscore prefix here? This is not intuitive to me, as we usually use this prefix for member vars. I would prefer a name that provides more context